src/Lumper.Lib/BSP/BspFile.cs

```diff
@@ -105,6 +105,15 @@ public sealed partial class BspFile : IDisposable
         if (path is null && FilePath is null)
             throw new ArgumentException("Not given a path to write to, and current BSP doesn't have a path");
 
+        if (Path.GetFileName(path) != FilePath)
+        { 
+            Dictionary<string, string> modified = GetLump<PakfileLump>().RenameDefaultCubemap(Path.GetFileName(path));
+            foreach (KeyValuePair<string,string> modifiedPath in modified)
+            {
+                GetLump<PakfileLump>().UpdatePathReferences(modifiedPath.Value, modifiedPath.Key);
```

Comment:
Weird, UpdatePathReferences doesn't live in Lumper.Lib, I suspect you missed something from your commits.